### PR TITLE
NIFI-7263 Add [No Tracking] Listing Strategy to ListFile/ListFTP proc…

### PR DESCRIPTION
**Pull request by walmut**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Add a new Listing Strategy called [No Tracking] to LisFile/ListFTP.  The purpose of this improvement is give a choice to list all files as like as GetFile processor does. As List/Fetch mechanism.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/list/AbstractListProcessor.java [hunk 1]
```diff
@@ -192,12 +192,16 @@
                     " However additional DistributedMapCache controller service is required and more JVM heap memory is used." +
                     " See the description of 'Entity Tracking Time Window' property for further details on how it works.");
 
+    public static final AllowableValue NO_TRACKING = new AllowableValue("none", "No Tracking",
+            "This strategy lists an entity without any tracking. Any property that related to persist state" +
+                    " will be disregarded.");
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/list/AbstractListProcessor.java]: I'd probably try to be even more descriptive here... I think it's worth saying that the same files will be listed at each execution of this processor, and that it's recommended to change the default run schedule value.

Comment (walmut) [reply nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/list/AbstractListProcessor.java]: @pvillard31 How about this: "This strategy lists an entity without any tracking. The same entity will be listed each time on executing this processor. It is recommended to change the default run schedule value. Any property that related to the persisting state will be disregarded."

**Comment by MarcSkovMadsen** [comment]
Any news on this one? I would also be interested in using this.

**Comment by humpfhumpf** [comment]
It is a mandatory feature in my opinion. @walmut could you please try to fix these conflicts?

**Comment by walmut** [comment]
@humpfhumpf the conflicts has been resolved.

**Comment by humpfhumpf** [comment]
> @humpfhumpf the conflicts has been resolved.

Thank you!
For an unknown reason (to me), github still detects a conflict on these 2 files:
```
nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/main/java/org/apache/nifi/processor/util/list/AbstractListProcessor.java
nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/test/java/org/apache/nifi/processor/util/list/TestAbstractListProcessor.java
```

**Comment by humpfhumpf** [comment]
Hi @pvillard31 ,
This PR is now stucked, in my opinion, on a CICD workflow error.
Could you please advise?

**Comment by joewitt** [comment]
This PR has been pretty difficult to get reviewed/merged because the commits create merge conflicts as noted previously and they then reference a merge commit which is an actual commit but that appears to alter the original commit.  It isn't clear to me how these three commits were even constructed.  Anyway, I've manually pulled in the first commit and modified to resolve the merge commits as best I can tell then pulled in the second commit.  The third commit then appears not necessary.  

Assuming that goes well and builds I'll do a quick manual test. If that is fine will merge.  Else will remove from this release for further effort.  I still don't quite follow the true benefit this brings for genuine use cases but as long as it is helpful for the case(s)/users noted on this thread then seems ok.

**Comment by joewitt** [comment]
ok manual testing shows it works as described.  It lists over and over with no tracking.  We'll have to keep an eye on this and see if it creates more confusion but otherwise hopefully it is helpful to those that wanted this.  Thanks for the contrib!

+1 merged to main

**Comment by humpfhumpf** [comment]
Thank you Joe.
The only "risk", I can imagine, is when the listed files are not (re)moving by FetchFile fast enough. Then, ListFile could find them again on next trigger/run. In my opinion, this is a Dataflow Manager responsability to choose this 'No Tracking' strategy.

**Pull request closed without merging**